homeassistant/components/synology_dsm/__init__.py

```diff
@@ -164,6 +163,7 @@ def _fetch_device_configuration(self):
         self.information = self.dsm.information
         self.utilisation = self.dsm.utilisation
         self.storage = self.dsm.storage
+        self.network = self.dsm.network
```

Comment:
This is the better way - the name of the DSM is available via network.hostname

Reply:
Yeay nice :+1:, indeed the network API :muscle: 

You can just store the hostname here, we not need the whole network object

Reply:
Getting the name from the device and not requiring it to configure or adding complexity to an options flow seems like a good solution.